app/api/data.ts

```diff
@@ -0,0 +1,112 @@
+import * as FileSystem from 'expo-file-system';
```

Comment:
Since interfacing with the Filesystem is costly (and thus async), it may be smarter to interface with something like `localStorage` by default (which is sync), and only write to Filesystem at the end. Up to you though.